components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -34,6 +34,12 @@ internal object Dispatchers {
                     block()
                 }
                 null
+            } else if (!Glean.isInitialized()) {
+                coroutineScope.launch {
+                    Glean.queueTaskForInit {
```

Comment:
We should do two things here, IMHO: 

- log that the call is being dalayed, due to Glean not being initialized;
- add an upper bound to the number of possible enqueued calls.